src/algorithms/models/edgecost.ts

```diff
@@ -7,18 +9,111 @@ class EdgeCost {
    * @param ecs EdgeCost[] costs to be combined
    * @return EdgeCost Combined cost of all ecs
    */
+
   static combine(...ecs: EdgeCost[]): EdgeCost {
     const res = new EdgeCost();
     for (const ec of ecs) {
       res.distance += ec.distance;
+      res.time += ec.time;
+      res.road_cost += ec.distance * ec.road_cost;
     }
+
+    res.road_cost /= res.distance;
+
     return res;
   }
 
   /**
    * Real distance between two vertices in meters
+   * Real time between two vertices in minutes
+   * Type of road between two vertices as a float value
+   */
+  distance: number;
+  time: number;
+  road_cost: number;
+
+  /**
+   * Constructor for the EdgeCost object which gives every edge between vertices a weight value for further calculation
+   * @param v1: Vertex - vertex object which forms the edge with the second one
+   * @param v2: Vertex - vertex which forms the edge with the first one
+   * @param road_type: string - type of road which connects the two vertices
    */
-  distance = 0;
+  constructor(v1: Vertex = null, v2: Vertex = null, road_type: string = null) {
+    if (v1 == null) {
+      return;
+    }
+
+    const lngSq = Math.pow((v1.lng - v2.lng), 2);
+    const latSq = Math.pow((v1.lat - v2.lat), 2);
+    this.distance = Math.sqrt(latSq + lngSq);
+
+    let max_speed: number;
+
+    switch (road_type) {
```

Comment:
Let's move the switch to a separate private method. So the constructor is not that looong.

Reply:
Done